eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1896,10 +1897,12 @@ public function deleteVersion(APIVersionInfo $versionInfo)
 
         $this->repository->beginTransaction();
         try {
-            $this->persistenceHandler->contentHandler()->deleteVersion(
+            $versionsCount === 1
+            ? $this->persistenceHandler->contentHandler()->deleteContent($versionInfo->contentInfo->id)
+            : $this->persistenceHandler->contentHandler()->deleteVersion(
```

Comment:
First of all, I don't feel comfortable making such an operation using a ternary operator. In this particular case I'd personally prefer a full fledged `if`.

Second, there seems to be a weird indentation issue here.

At the very least fix the indentation, but I think that others will agree that due to the complexity of the operation called, it should be an `if` statement.

Reply:
I don't mind ternary operator, but if/else may be a bit more appropriate. In either case I would expose content handler as a local variable to avoid repeated calls.